### DISCUSSION TITLE
File download statistics (number of downloads)

### DISCUSSION THREAD

**Discussion by LeBogoo** [pocketbase/pocketbase#4214] [Q&A] [answered]
Is there any way to know how many times a particular file has been downloaded?

If not, do you have a suggestion on how I can start to write my own Pocketbase extension that logs downloads and makes them available at a new URL?
My idea would be to store the data in the .attrs files and retrieve them when the count is requested.

I'm pretty new to Pocketbase, so I don't have that much experience yet, but I'm willing to learn how to build something like this.

**Comment by ganigeorgiev** [comment, marked as answer]
You can attach to the [`OnFileDownloadRequest` hook](https://pocketbase.io/docs/go-event-hooks/#onfiledownloadrequest) with Go or JS and increment a `number` field programatically.
For the latter, you could check the JS example in https://github.com/pocketbase/pocketbase/discussions/4012#discussioncomment-7932593.

**Reply by LeBogoo** [reply]
Thank you so much! I appreciate the help!

**Reply by LeBogoo** [reply]
It works perfectly.
Here is my complete main.pb.js file:

```js
/// <reference path="../pb_data/types.d.ts" />

onFileDownloadRequest((event) => {
  console.log("Downlaoded mod " + event.record.get("name") + ". Incrementing download count.");

  $app.dao().runInTransaction((txDao) => {

    let record = event.record;
    record.set("downloads", record.get("downloads") + 1);
    txDao.saveRecord(record)
  });
});
```
